test/Microsoft.TemplateEngine.IDE.IntegrationTests/End2EndTests.cs

```diff
@@ -66,12 +68,35 @@ internal async Task ValueForms_DerivedSymbolFromGeneratedSymbolTest()
             var foundTemplates = await bootstrapper.GetTemplatesAsync(new[] { WellKnownSearchFilters.NameFilter("TestAssets.ValueForms.DerivedSymbolFromGeneratedSymbol") }).ConfigureAwait(false);
             var result = await bootstrapper.CreateAsync(foundTemplates[0].Info, "Real.Web.App", output, new Dictionary<string, string?>()).ConfigureAwait(false);
 
-            Assert.Equal(Edge.Template.CreationResultStatus.Success, result.Status);
+            Assert.Equal(CreationResultStatus.Success, result.Status);
 
             string targetFile = Path.Combine(output, "Real.Web.App.txt");
             Assert.True(File.Exists(targetFile));
 
             await Verify(File.ReadAllText(targetFile));
         }
+
+        [Fact]
+        internal async Task Test_CreateAsync_OnInvalidParamsPassed()
+        {
+            using Bootstrapper bootstrapper = GetBootstrapper();
+            string templateLocation = GetTestTemplateLocation("TemplateWithPortsAndCoalesce");
+            await InstallTemplateAsync(bootstrapper, templateLocation).ConfigureAwait(false);
+
+            string output = TestUtils.CreateTemporaryFolder();
+
+            IReadOnlyList<ITemplateMatchInfo> foundTemplates = await bootstrapper.GetTemplatesAsync(new[] { WellKnownSearchFilters.NameFilter("TestAssets.TemplateWithPortsAndCoalesce") }).ConfigureAwait(false);
+
+            Dictionary<string, string?> parameters = new()
+            {
+                { "userPort1", "non-int" },
+                { "userPort2", string.Empty }
+            };
+
+            ITemplateCreationResult result = await bootstrapper.CreateAsync(foundTemplates[0].Info, "Test", output, parameters).ConfigureAwait(false);
+
+            Assert.Equal(CreationResultStatus.InvalidParamValues, result.Status);
+            Assert.Equal("userPort1, userPort2", result.ErrorMessage);
+        }
```

Comment:
@phenning I added this test to verify behavior of type mismatch and it looks ok-ish for me. Error message could be better, we have https://github.com/dotnet/templating/issues/4225 for that. Could it be that VS host is ignoring `CreationResultStatus` here?